fuzzing/private/engine.bzl

```diff
@@ -44,15 +43,20 @@ def _cc_fuzzing_engine_impl(ctx):
         launcher_runfiles = launcher_runfiles.merge(ctx.runfiles(files = data_files))
         launcher_runfiles = launcher_runfiles.merge(data_label[DefaultInfo].default_runfiles)
 
-    cc_library_info = ctx.attr.library[CcInfo]
-    cc_fuzzing_engine_info = CcFuzzingEngineInfo(
+    fuzzing_engine_info = FuzzingEngineInfo(
         display_name = ctx.attr.display_name,
-        engine_library_info = cc_library_info,
         launcher = ctx.file.launcher,
         launcher_runfiles = launcher_runfiles,
         launcher_environment = env_vars,
     )
-    return [ctx.attr.library[DefaultInfo], cc_library_info, cc_fuzzing_engine_info]
+    return [
+        ctx.attr.library[DefaultInfo],
+        ctx.attr.library[language_provider],
+        fuzzing_engine_info,
+    ]
```

Comment:
Maybe we can push the return of this vector in the top-level _impl function? This way, we can achieve a better separation of concerns and have this function only compute and return the FuzzingEngineInfo provider (could call it, e.g., `_get_fuzzing_engine_info`), with no need to forward the language-specific provider.

Reply:
Right, that's better. I pushed new commits that do this.